app/upgrades/v8/upgrades.go

```diff
@@ -49,6 +51,45 @@ func FixBankMetadata(ctx sdk.Context, keepers *keepers.AppKeepers) error {
 	return nil
 }
 
+func QuicksilverFix(ctx sdk.Context, keepers *keepers.AppKeepers) error {
+	// Refund stuck coins from ica address
+	sourceAddress, err := sdk.AccAddressFromBech32("cosmos13dqvh4qtg4gzczuktgnw8gc2ewnwmhdwnctekxctyr4azz4dcyysecgq7e")
+	if err != nil {
+		return errors.New("invalid source address")
+	}
+	destinationAddress, err := sdk.AccAddressFromBech32("cosmos1jc24kwznud9m3mwqmcz3xw33ndjuufnghstaag")
+	if err != nil {
+		return errors.New("invalid destination address")
+	}
+
+	// Get balance from stuck address and subtract 1 uatom sent by bad actor
+	sourceBalance := keepers.BankKeeper.GetBalance(ctx, sourceAddress, "uatom")
+	if sourceBalance.IsGTE(sdk.NewCoin("uatom", sdk.NewInt(1))) {
+		refundBalance := sourceBalance.SubAmount(sdk.NewInt(1))
+		err = keepers.BankKeeper.SendCoins(ctx, sourceAddress, destinationAddress, sdk.NewCoins(refundBalance))
+		if err != nil {
+			return errors.New("unable to refund coins")
+		}
```

Comment:
@stana-ethernal , since we might skip `sendCoins` if source does not have balance. maybe we can log here that the fund is sent back , so that we can make sure in mainnet and theta-testnet, we have this step executed